casper/src/main/scala/coop/rchain/casper/util/TopologicalSortUtil.scala

```diff
@@ -0,0 +1,26 @@
+package coop.rchain.casper.util
+
+import coop.rchain.casper.Estimator.BlockHash
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.casper.util.ProtoUtil.blockNumber
+
+object TopologicalSortUtil {
+  type BlockSort = Vector[Vector[BlockHash]]
+  def update(sort: BlockSort, offset: Long, block: BlockMessage): BlockSort = {
+    val hash   = block.blockHash
+    val number = (blockNumber(block) - offset).toInt
```

Comment:
You should probably check this conversion on errors (e.g. if it fits to `Int` and it actually greater than 0.